src/test/java/com/yelp/nrtsearch/server/luceneserver/search/collectors/additional/StringTermsCollectorManagerTest.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.search.collectors.additional;
+
+import com.yelp.nrtsearch.server.grpc.AddDocumentRequest;
+import com.yelp.nrtsearch.server.grpc.FieldDefRequest;
+import com.yelp.nrtsearch.server.grpc.SearchResponse;
+import com.yelp.nrtsearch.server.grpc.TermsCollector;
+import io.grpc.testing.GrpcCleanupRule;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class StringTermsCollectorManagerTest extends TermsCollectorManagerTestsBase {
```

Comment:
The string collector is only used in the case where the doc values are `BINARY`, otherwise ordinals are aggregated instead. This currently only happens for single value text https://github.com/Yelp/nrtsearch/blob/master/src/main/java/com/yelp/nrtsearch/server/luceneserver/field/TextBaseFieldDef.java#L91 We may want to make ordinal vs binary text definitions more explicit in the future.